### PR TITLE
Adding license field to GithubInfo and GithubModel

### LINKED ISSUE #1021: Add Project license from Github
### Motivation

One can configure the license of its project in its Github repository. This information should appear in the Scaladex project page.

### Identified Obstacles

_No response_

### Implementation Guideline

1. Add a `licence: Option[License]` field in `GithubInfo`
2. Add `licenseName: Option[String]` field in the `GithubModel.Repository` and fill it using the `GithubModel.Repository` decoder. The schema of the json response can be found in https://docs.github.com/en/rest/repos/repos#get-a-repository
3. Use the `LicenseCleanup` to get a the `License` object from the name.
4. Make the license appear in the header of the project page.

### Expectations

- [x] Get the License from Github
- [ ] The license appears in the header of the project page, as a link to the license page.

### PR DESCRIPTION
**Pull request by jyoo980**
Now extracting the `spdx_id` from the JSON response from GitHub as
`Option[String]`, assigned to field `licenseName` in `GithubModel.Repository`.
In order to extract a value of type `Option[License]` from this, there needs
to be a way for `GithubClient` to get a hold of `LicenseCleanup`.

I don't think there's a clean way to do this right now w/o injecting the
`LicenseCleanup` instance directly, or passing down the `DataPath`. I don't
want to do this because it seems like these two components shouldn't really be
tightly coupled together.

Resolves #1021

### CODE REVIEW

File: modules/infra/src/main/scala/scaladex/infra/GithubClientImpl.scala [hunk 1]
```diff
@@ -103,7 +103,8 @@ class GithubClientImpl(token: Secret)(implicit val system: ActorSystem)
       codeOfConduct = communityProfile.flatMap(_.codeOfConductFile).map(Url),
       chatroom = chatroom.map(Url),
       openIssues = openIssues.map(_.toGithubIssue).toList,
-      scalaPercentage = Option(scalaPercentage)
+      scalaPercentage = Option(scalaPercentage),
+      license = None // Need to somehow get access to the `LicenseCleanup` here.
```
[the remark this answers is not in the document]
Comment (adpi2) [reply modules/infra/src/main/scala/scaladex/infra/GithubClientImpl.scala:107]: :thinking: I am not concerned about using `LicenseCleanup` here. `GithubClientImpl` is an interface between the Github API and our internal model, and `LicenseCleanup` is merely a map of IDs to `License`. Even if it reads the map from the filesystem, I think we should consider it as being just static.

[the remark this answers is not in the document]
Comment (jyoo980) [reply modules/infra/src/main/scala/scaladex/infra/GithubClientImpl.scala:107]: Sure, that works.

However, it looks like `LicenseCleanup` lives in the `data` module, which has the `infra` module as a dependency, i.e., I can't import it without introducing a circular dependency.

A way I could resolve this is by creating a new `common` module, but that seems like overkill. Do you have any thoughts?

[the remark this answers is not in the document]
Comment (adpi2) [reply modules/infra/src/main/scala/scaladex/infra/GithubClientImpl.scala:107]: Ah right!

This should solve the issue: https://github.com/scalacenter/scaladex/pull/1034

I moved the `LicenseCleanup` logic to the `License` companion object. So you should be able to just do `License.get(name)` now.

**Comment by jyoo980** [comment]
@adpi2 Sorry for the delay, been super busy. I think this is ready for another look!

**Review by adpi2 — APPROVED** [review]
Thank you for this contribution

**Pull request merged**